great_expectations/dataset/base.py

```diff
@@ -1032,18 +1032,22 @@ def _format_column_map_output(self,
             return return_obj
 
         # Try to return the most common values, if possible.
-        try:
-            partial_unexpected_counts = [
-                {'value': key, 'count': value}
-                for key, value
-                in sorted(
-                    Counter(unexpected_list).most_common(
-                        result_format['partial_unexpected_count']),
-                    key=lambda x: (-x[1], x[0]))
-            ]
-        except TypeError:
-            partial_unexpected_counts = [
-                'partial_exception_counts requires a hashable type']
+        # If we have a dict, we probably had a dataframe; punt
+        if isinstance(unexpected_list, list):
```

Comment:
I don't fully understand the case this traps. I'll look in the tests...

Reply:
I didn't see a case that raises the `partial_exception_counts requires a hashable type` behavior...

Reply:
The `partial_exception_counts requires a hashable type` notation is not new (that case can arise if you have complex values that cannot be counted by Counter).

The new case was when unexepected_list was not a list (i.e. because I was using the default dict view that preserved indices). Switching to the records view obviates that check, so I'm just reverting this hunk.